### PR TITLE
Automatically detect extras from the PyPI metadata

### PR DESCRIPTION
**Pull request by befeleme**

### CODE REVIEW

File: pyp2spec/pyp2conf.py [hunk 1]
```diff
@@ -220,6 +220,27 @@ def archive_name(self, version):
...
+        Packages define extras explicitly via `Provides-Extra` and
+        indirectly via `Requires-Dist` metadata.
+        PyPI metadata doesn't provide the first one, but it is possible to
+        derive extras names from the `requires_dist` key.
+        Example value: ["sphinxcontrib-websupport ; extra == \"docs\"",
+        "flake8>=3.5.0 ; extra == \"lint\""]
+        """
+
+        extras = set()
+        requires_dist = self.package_data["info"]["requires_dist"]
```
Comment (hroncok) [inline pyp2spec/pyp2conf.py:235]: This also suffers from https://github.com/befeleme/pyp2spec/issues/41

Comment (befeleme) [reply pyp2spec/pyp2conf.py:238]: Fixed in https://github.com/befeleme/pyp2spec/pull/42

**Comment by befeleme** [comment]
That was a little too big of a rework to just throw a bunch of fixups. Rebased.

**Comment by befeleme** [comment]
TODO: some tests for the `extras` parsing from the metadata.

File: pyp2spec/template.spec [hunk 2]
```diff
@@ -29,14 +29,21 @@ BuildRequires:  {{br}}
...
+{% endif %}
 
 %prep
 %autosetup -p1 -n {{archive_name}}-{{pypi_version}}
 
 
 %generate_buildrequires
-%pyproject_buildrequires{% if extra_build_requires %} {{extra_build_requires}}{% endif %}
+{% if extras -%}
+# Keep only those extras which you actually want to package
```
Comment (hroncok) [inline pyp2spec/template.spec:44]:
```suggestion
# Keep only those extras which you actually want to package or use during tests
```

**Pull request merged**